demo.py

```diff
@@ -51,7 +56,7 @@ def main():
         "product_matchrules": [
             ["CREATE", "foo.tar.gz"],
         ],
-        "pubkeys": [bob_key["keyid"]],
+        "pubkeys": [bob_public["keyid"]],
```

Comment:
Instead of writing the layout itself, instantiate Link and Inspection objects to add to a Layout object and then dump it. Maybe it's a little bit more tedious in the short term, but should be less "volatile" (and easier to update if something breaks).

I don't think this is anything major though.